sql-plugin/src/main/scala/com/nvidia/spark/rapids/GpuOrcScan.scala

```diff
@@ -234,8 +234,10 @@ object GpuOrcScan extends Arm {
         withResource(col.castTo(DType.INT64)) { longs =>
           // In CPU, ORC assumes the integer value is in seconds, and returns timestamp in
           // micro seconds.
-          withResource(longs.bitCastTo(DType.TIMESTAMP_SECONDS)) { seconds =>
-            seconds.castTo(DType.TIMESTAMP_MICROSECONDS)
+          withResource(Scalar.fromLong(1e6.toLong)) { value =>
```

Comment:
We don't want to use floating point and force a cast at runtime to a long.  Just use a long, either `1000000L` or `TimeUnit.SECONDS.toMicros(1)` if you want to go that route.